include/sound/sof/ipc4/header.h

```diff
@@ -0,0 +1,324 @@
+#define SOF_IPC4_GLB_PIPE_MEM_SIZE(x)		((x) << SOF_IPC4_GLB_PIPE_MEM_SIZE_SHIFT)
+
+#define SOF_IPC4_GL_PIPE_EXT_LP_SHIFT		0
+#define SOF_IPC4_GL_PIPE_EXT_LP_MASK		BIT(0)
+#define SOF_IPC4_GL_PIPE_EXT_LP(x)		((x) << SOF_IPC4_GL_PIPE_EXT_LP_SHIFT)
+
+/* pipeline set state ipc msg */
+#define SOF_IPC4_GL_PIPE_STATE_TYPE_SHIFT	24
+#define SOF_IPC4_GL_PIPE_STATE_TYPE_MASK	GENMASK(28, 24)
+#define SOF_IPC4_GL_PIPE_STATE_TYPE(x)		((x) << SOF_IPC4_GL_PIPE_STATE_TYPE_SHIFT)
+
+#define SOF_IPC4_GL_PIPE_STATE_ID_SHIFT		16
+#define SOF_IPC4_GL_PIPE_STATE_ID_MASK		GENMASK(23, 16)
+#define SOF_IPC4_GL_PIPE_STATE_ID(x)		((x) << SOF_IPC4_GL_PIPE_STATE_ID_SHIFT)
+
+#define SOF_IPC4_GL_PIPE_STATE_SHIFT		0
+#define SOF_IPC4_GL_PIPE_STATE_MASK		GENMASK(15, 0)
+#define SOF_IPC4_GL_PIPE_STATE(x)		((x) << SOF_IPC4_GL_PIPE_STATE_SHIFT)
+
+enum sof_ipc4_sampling_frequency {
+	SOF_IPC4_FS_8000HZ	= 8000,
+	SOF_IPC4_FS_11025HZ	= 11025,
+	SOF_IPC4_FS_12000HZ	= 12000, /* Mp3, AAC, SRC only. */
+	SOF_IPC4_FS_16000HZ	= 16000,
+	SOF_IPC4_FS_18900HZ	= 18900, /* SRC only for 44100 */
+	SOF_IPC4_FS_22050HZ	= 22050,
+	SOF_IPC4_FS_24000HZ	= 24000, /* Mp3, AAC, SRC only. */
+	SOF_IPC4_FS_32000HZ	= 32000,
+	SOF_IPC4_FS_37800HZ	= 37800, /* SRC only for 44100 */
+	SOF_IPC4_FS_44100HZ	= 44100,
+	SOF_IPC4_FS_48000HZ	= 48000, /* Default. */
+	SOF_IPC4_FS_64000HZ	= 64000, /* AAC, SRC only. */
+	SOF_IPC4_FS_88200HZ	= 88200, /* AAC, SRC only. */
+	SOF_IPC4_FS_96000HZ	= 96000, /* AAC, SRC only. */
+	SOF_IPC4_FS_176400HZ	= 176400, /* SRC only. */
+	SOF_IPC4_FS_192000HZ	= 192000, /* SRC only. */
+	SOF_IPC4_FS_INVALID
+};
+
+enum sof_ipc4_bit_depth {
+	SOF_IPC4_DEPTH_8BIT	= 8, /* 8 bits depth */
+	SOF_IPC4_DEPTH_16BIT	= 16, /* 16 bits depth */
+	SOF_IPC4_DEPTH_24BIT	= 24, /* 24 bits depth - Default */
+	SOF_IPC4_DEPTH_32BIT	= 32, /* 32 bits depth */
+	SOF_IPC4_DEPTH_64BIT	= 64, /* 64 bits depth */
+	SOF_IPC4_DEPTH_INVALID
+};
+
+enum sof_ipc4_channel_config {
+	/* one channel only. */
+	SOF_IPC4_CHANNEL_CONFIG_MONO			= 0,
+	/* L & R. */
+	SOF_IPC4_CHANNEL_CONFIG_STEREO			= 1,
+	/* L, R & LFE; PCM only. */
+	SOF_IPC4_CHANNEL_CONFIG_2_POINT_1		= 2,
+	/* L, C & R; MP3 & AAC only. */
+	SOF_IPC4_CHANNEL_CONFIG_3_POINT_0		= 3,
+	/* L, C, R & LFE; PCM only. */
+	SOF_IPC4_CHANNEL_CONFIG_3_POINT_1		= 4,
+	/* L, R, Ls & Rs; PCM only. */
+	SOF_IPC4_CHANNEL_CONFIG_QUATRO			= 5,
+	/* L, C, R & Cs; MP3 & AAC only. */
+	SOF_IPC4_CHANNEL_CONFIG_4_POINT_0		= 6,
+	/* L, C, R, Ls & Rs. */
+	SOF_IPC4_CHANNEL_CONFIG_5_POINT_0		= 7,
+	/* L, C, R, Ls, Rs & LFE. */
+	SOF_IPC4_CHANNEL_CONFIG_5_POINT_1		= 8,
+	/* one channel replicated in two. */
+	SOF_IPC4_CHANNEL_CONFIG_DUAL_MONO		= 9,
+	/* Stereo (L,R) in 4 slots, 1st stream: [ L, R, -, - ] */
+	SOF_IPC4_CHANNEL_CONFIG_I2S_DUAL_STEREO_0	= 10,
+	/* Stereo (L,R) in 4 slots, 2nd stream: [ -, -, L, R ] */
+	SOF_IPC4_CHANNEL_CONFIG_I2S_DUAL_STEREO_1	= 11,
+	/* L, C, R, Ls, Rs & LFE., LS, RS */
+	SOF_IPC4_CHANNEL_CONFIG_7_POINT_1		= 12,
+	SOF_IPC4_CHANNEL_CONFIG_INVALID
```

Comment:
same here, 13 is invalid?

Reply:
> same here, 13 is invalid?

Those invalid enum values are copied from the existing cavs firmware. It will be mismatched if we change it. The question is that do we need to be compatible with the existing cavs firmware? Or be compatible with the converged firmware is good enough?  @plbossart @lgirdwood